File: Project_24/frontend_bkup/lib/models/create_habit_request.dart
import 'dart:convert';

class CreateHabitRequest {
  final int userId;
  final int habitCategoryId;
  final String name;
  final String description;
  final DateTime startDate;
  final DateTime? endDate;
  final String frequency;
  final DateTime reminderTime;
  final bool notificationsEnabled;
  final String iconName;
  final String colorHex;

  const CreateHabitRequest({
    required this.userId,
    required this.habitCategoryId,
    required this.name,
    required this.description,
    required this.startDate,
    this.endDate,
    required this.frequency,
    required this.reminderTime,
    required this.notificationsEnabled,
    required this.iconName,
    required this.colorHex,
  });

  Map<String, dynamic> toJson() {
    return {
      "userId": userId,
      "habitCategoryId": habitCategoryId,
      "name": name,
      "description": description,
      "startDate": _date(startDate),
      "endDate": endDate == null ? null : _date(endDate!),
      "frequency": frequency,
      "reminderTime": _time(reminderTime),
      "notificationsEnabled": notificationsEnabled,
      "iconName": iconName,
      "colorHex": colorHex,
    };
  }

  String toRawJson() => jsonEncode(toJson());

  String _date(DateTime date) {
    return date.toIso8601String().split("T").first;
  }

  String _time(DateTime time) {
    final h = time.hour.toString().padLeft(2, '0');
    final m = time.minute.toString().padLeft(2, '0');
    return "$h:$m:00";
  }
}

File: Project_24/LifeTracker/backend/src/main/java/com/lifetracker/modules/habits/dto/CreateHabitRequest.java
package com.lifetracker.modules.habits.dto;

import com.lifetracker.modules.habits.enums.HabitFrequency;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.time.LocalTime;

public record CreateHabitRequest(
        @NotNull Long habitCategoryId,
        @NotBlank String name,
        String description,
        @NotNull LocalDate startDate,
        LocalDate endDate,
        @NotNull HabitFrequency frequency,
        @NotNull LocalTime reminderTime,
        @NotNull Boolean notificationsEnabled,
        String iconName,
        String colorHex
) {
}
